[fix] Transacciones coherentes al hacer pagos con descuentos y comisiones

raunavcru opened the pull request.

raunavcru committed:
[fix] Transaction anterior

Commit:
[WIP] Creación de transacciones correctamente

Commit:
[refactor][WIP] Inscripción con descuento con un solo controlador

angelcalzadollamas commented:
Un inciso con esto, he visto el último commit de esta pull y se tocan cosas de la vista detail event. Como dije ayer esta vista se ha cambiado y está en una pull por lo que podrían darse conflictos en una rama o en otra dependiendo del cuál mergee primero.

Solo para avisar y que no se piense que alguien ha tocado algo.

Commit:
[refactor][WIP] Cambio modelo de transacción

Ahora se guarda la comisión y el descuento

Commit:
[refactor][WIP] Se quitan eventpoints una vez se hace inscripción

Commit:
[feat][WIP] Se elimina transacción y devuelven eventpoints al rechazar
inscripción

Commit:
[feat][WIP] Se devuelven eventpoints al cancelar asistencia

Commit:
[feat][WIP] Se eliminan transacciones y se devuelven points al eliminar
evento

Commit:
[fix] Quitado código comentado

Commit:
[fix] Se devuelven eventpoints al cancelar evento si este no ha empezado

Al igual que se eliminan las transacciones

Commit:
[fix] Se carga la comisión correcta al hacer el cobro

Commit:
[refactor] Sólo se muestran las transacciones efectuadas

Es decir las transacciones que tiene el atributo is_paid_for a True

Commit:
[fix] Error al cancelar asistencia a evento por no usar iterable

Se estaba pasando un sólo usuario y se intentaba iterar sobre él dando
error

Commit:
[fix] Calculo de descuento devuelve negativo

Commit:
[refactor] Añadidas comisiones de stripe como variables a setting

Commit:
[fix] Se calcula el bonus por asistencia con el precio en centimos

Esto es así porque el valor del eventpoint está establecido en centimos
para facilitar el uso de strip

angbongon committed:
Merge branch 'dev' into fix/be/CU-31

angelcalzadollamas reviewed (changes requested):
En este mensaje voy a mostrar las casuísticas que yo he probado para que veamos si me dejo algo atrás o no. Están marcados aquellos checbox de comprobaciones que van bien

**Comprobaciones**

[] 1.  Usuario se inscribe a evento con 0 eventpoints y sin tarjeta de crédito registrada.
    _La funcionalidad con respecto a las transacciones y eventpoints va pero el js de stripe no muestra el costo total de la entrada con la comision. Dejalo como estaba antes. Imagen de como se ve ahora:_
![1-El js de stripe no muestra el total a pagar](https://user-images.githubusercontent.com/28622424/79288805-fdcec680-7ec7-11ea-82f3-cb3c42e93e48.PNG)

[x] 2. Usuario se inscribe a evento sin eventpoints con tarjeta registrada.
[] 3. Usuario se inscribe a evento con eventpoints sin tarjeta registrada.
    _La funcionalidad con respecto a las transacciones y evenpoints van bien pero el css se ha roto. No le pongas en este caso a los botones el paréntesis mostrándo el precio ya que como luego sale el js de stripe, se va a acabar viendo. Aquí una imagen de como se ve ahora:_
![La disposición de los botones está mal](https://user-images.githubusercontent.com/28622424/79288950-6027c700-7ec8-11ea-9444-5e543d82f7d3.PNG)

[x] 4. Usuario se inscribe a evento con eventpoints y tarjeta registrada.
[x] 5. Anfitrión recibe pago de una transacción sin descuento.
[x] 6. Anfitrión recibe pago de una transacción con descuento.
[x] 7. Asistente cancela su asistencia con enrollment en estado pending a más de 3 días de la celebración.
[x] 8. Asistente cancela su asistencia con enrollment en estado pending a 3 días de la celebración.
[x] 9. Asistente cancela su asistencia con enrollment en estado accepted a más de 3 días de la celebración.
[x] 10. Asistente cancela su asistencia con enrollment en estado accepted a 3 días de la celebración.
[x] 11. Anfitrión cancela su evento a más de 4 días de la celebración.
[x] 12. Anfitrión cancela su evento a 4 días de la celebración.

> Me faltaría probar el olvido, pero como te dije, debes mergear dev a esta rama porque la tienes un poco antiguada dado que me ha dado un error que recuerdo se arregló en dev.

Commit:
[fix] Se muestra bien la cantidad de stripe checkout cuando 0 eventpoint

Commit:
[feat][refactor] Al eliminar usuario se devuelven eventpoints

Se devuelven los evenpoints a todos los usuarios inscritos a los eventos
del usuario a eliminar que son en el futuro

angbongon commented:
@angcallla en principio ya está implementado para cuando se elimina un usuario. Te recomiendo que a la hora de probarlo no uses el populate al completo, populalo sólo con usuarios, crea con cualquiera un evento y haz que unos pocos se inscriban (no todos para comprobar que se toque sólo a los que se deben tocar) y haz las pruebas. El tema de como se ve el modal cuando hace falta credit card no lo he tocado para que sea Juanmi el que lo modifique cuando suba la vista a dev.

angelcalzadollamas reviewed (changes requested):
Ya te puedo enviar esto. No he podido revisar nada porque cuando un usuario tiene eventpoints, a la hora de inscribirse a un evento, no me da la opción de usarlos por lo que no puedo revisar lo referente a las transacciones y de más cosas.

Creo que llegados a este punto lo mejor será que me espere a que lo tengas todo listo junto a lo de juanmi para ya revisarlo todo en su totalidad

Commit:
[fix] No aparece el modal con las opciones de paga con eventpoints

La variable que controlaba dicho comportamiento estaba borrada

angbongon commented:
> Ya te puedo enviar esto. No he podido revisar nada porque cuando un usuario tiene eventpoints, a la hora de inscribirse a un evento, no me da la opción de usarlos por lo que no puedo revisar lo referente a las transacciones y de más cosas.
> 
> 
> 
> Creo que llegados a este punto lo mejor será que me espere a que lo tengas todo listo junto a lo de juanmi para ya revisarlo todo en su totalidad

Ya está arreglado. La variable que controlaba eso se había borrado.

angbongon committed:
Merge branch 'dev' into fix/be/CU-31

Commit:
[fix] Vars usadas en front no existían

angelcalzadollamas reviewed (changes requested):
**Comprobaciones**

[x] 1.  Usuario se inscribe a evento con 0 eventpoints y sin tarjeta de crédito registrada.
[x] 2. Usuario se inscribe a evento sin eventpoints con tarjeta registrada.
[] 3. Usuario se inscribe a evento con eventpoints sin tarjeta registrada.
    _La funcionalidad con respecto a las transacciones y evenpoints van bien pero el css se ha roto. Lo primero la disposición de los botones ya no están bien y lo segundo en este caso en concreto lo que te señalo en amarillo no debería aparecer. Imagen:_
![caso 3](https://user-images.githubusercontent.com/28622424/79482967-dedc4b80-8011-11ea-80a3-04a1b2eb34b0.PNG)

[] 4. Usuario se inscribe a evento con eventpoints y tarjeta registrada.
   _La funcionalidad con respecto a las transacciones y los eventpoints están bien pero el css se ha roto. Lo que te marco en amarillo no debería estar. Imagen:_
![Caso 4](https://user-images.githubusercontent.com/28622424/79483785-0384f300-8013-11ea-8d81-8a82e86ad56d.PNG)

[x] 5. Anfitrión recibe pago de una transacción sin descuento.
[x] 6. Anfitrión recibe pago de una transacción con descuento.
[x] 7. Asistente cancela su asistencia con enrollment en estado pending a más de 3 días de la celebración.
[x] 8. Asistente cancela su asistencia con enrollment en estado pending a 3 días de la celebración.
[x] 9. Asistente cancela su asistencia con enrollment en estado accepted a más de 3 días de la celebración.
[x] 10. Asistente cancela su asistencia con enrollment en estado accepted a 3 días de la celebración.
[] 11. Anfitrión cancela su evento con enrollment de asistente a PENDING.
   _Me he inscrito a un evento, luego me he logueado como el anfitrión y lo he cancelado y al asistente no se les ha devuelto los eventpoints._
[x] 12. Anfitrión cancela su evento con enrollment a ACCEPTED.
[x] 13. Anfitrión rechaza enrollment.
[] 14. Anfitrión olvida sus datos.
    _No se olvidan los datos, sigo estando logueado como el usuario_

Commit:
[fix] Código comentado para eliminar usuario

Commit:
[fix] Al eliminar evento se devuelven eventpoints a TODOS los inscritos

Es decir a todos lo que tengan una inscripcción pendiente o aceptada

juanmiguelruiz committed:
[fix] Integración estilo con funcionalidad

Commit:
[fix] Al eliminar usuario se devuelven eventpoints de TODOS los inscrito

Commit:
[fix] Selector para obtener TODOS los usuarios inscritos corregido

Al usar exclude tras usar filter no se estaban cogiendo correctamente
los usuarios

angelcalzadollamas committed:
Merge branch 'dev' into fix/be/CU-31
